src/main/java/org/snomed/snowstorm/mrcm/MRCMLoader.java

```diff
@@ -18,116 +21,159 @@
-						RuleStrength.lookupByConceptId(member.getAdditionalField("ruleStrengthId")),
-						ContentType.lookupByConceptId(member.getAdditionalField("contentTypeId"))
-				));
-			}
-		}
-		timer.checkpoint("Load attribute domains");
-
-
-		List<ReferenceSetMember> attributeRangeMembers = memberService.findMembers(branchPath, branchCriteria,
-				new MemberSearchRequest().referenceSet(Concepts.REFSET_MRCM_ATTRIBUTE_RANGE_INTERNATIONAL), LARGE_PAGE).getContent();
-
-		List<AttributeRange> attributeRanges = new ArrayList<>();
-		for (ReferenceSetMember member : attributeRangeMembers) {
-			// id	effectiveTime	active	moduleId	refsetId	referencedComponentId	rangeConstraint	attributeRule	ruleStrengthId	contentTypeId
-			if (member.isActive()) {
-				attributeRanges.add(new AttributeRange(
-						member.getMemberId(),
-						member.getEffectiveTime(),
-						member.isActive(),
-						member.getReferencedComponentId(),
-						member.getAdditionalField("rangeConstraint"),
-						member.getAdditionalField("attributeRule"),
-						RuleStrength.lookupByConceptId(member.getAdditionalField("ruleStrengthId")),
-						ContentType.lookupByConceptId(member.getAdditionalField("contentTypeId"))
-				));
-			}
-		}
-		timer.checkpoint("Load attribute ranges");
-
-		return new MRCM(domains, attributeDomains, attributeRanges);
-	}
-
-	private Constraint getConstraint(ReferenceSetMember mrcmMember, String fieldName) {
-		String constraint = mrcmMember.getAdditionalField(fieldName);
-		if (constraint == null || constraint.isEmpty()) {
-			logger.warn("No constraint found for '{}' in member {}.", fieldName, mrcmMember.getMemberId());
-			return null;
-		}
-		ExpressionConstraint ecl = eclQueryBuilder.createQuery(constraint);
-		if (ecl instanceof SubExpressionConstraint) {
-			SubExpressionConstraint sub = (SubExpressionConstraint) ecl;
-			return new Constraint(constraint, sub.getConceptId(), sub.getOperator());
-		} else if (ecl instanceof RefinedExpressionConstraint) {
-			RefinedExpressionConstraint refined = (RefinedExpressionConstraint) ecl;
-			SubExpressionConstraint sub = refined.getSubexpressionConstraint();
-			return new Constraint(constraint, sub.getConceptId(), sub.getOperator());
-		} else {
-			logger.error("Unable to process MRCM constraint '{}' in member {}.", constraint, mrcmMember.getMemberId());
-		}
-		return null;
-	}
+public class MRCMLoader implements CommitListener {
+    private static final Logger LOGGER = LoggerFactory.getLogger(MRCMLoader.class);
+
+    private final Map<String, MRCM> cache = new HashMap<>();
+
+    @Autowired
+    private ECLQueryBuilder eclQueryBuilder;
+
+    @Autowired
+    private ReferenceSetMemberService memberService;
+
+    @Override
+    public void preCommitCompletion(final Commit commit) throws IllegalStateException {
+        if (!this.cache.isEmpty()) {
+            final Branch branch = commit.getBranch();
+            final String branchPath = branch.getPath();
+
+            this.cache.remove(branchPath);
```

Comment:
Try to strike a balance between declaring variables and just calling getters to provide method parameters in a single line. Declaring variable can make code more readable if there is lots going on and you are trying to make a point and draw attention. This is usually when a variable will be used more than once. 
In this case I don't think any variables are needed, this would be fine `this.cache.remove(commit. getBranch(). getPath())`? In fact that could be the only line of this method, if block does not add anything : )